backend/src/main/kotlin/blue/mild/covid/vaxx/service/PatientService.kt

```diff
@@ -37,20 +37,34 @@ class PatientService(
     /**
      * Returns single patient with given personal number or throws exception.
      */
-    suspend fun getPatientByPersonalNumber(patientPersonalNumber: String): PatientDtoOut =
+    private suspend fun getPatientByPersonalNumber(patientPersonalNumber: String): PatientDtoOut? =
         patientRepository.getAndMapPatientsBy {
             Patients.personalNumber eq patientPersonalNumber.normalizePersonalNumber()
         }.singleOrNull()?.withSortedAnswers()
-            ?: throw entityNotFound<Patients>(Patients::personalNumber, patientPersonalNumber)
 
     /**
      * Returns single patient with given insurance number or throws exception.
      */
-    suspend fun getPatientByInsuranceNumber(patientInsuranceNumber: String): PatientDtoOut =
+    private suspend fun getPatientByInsuranceNumber(patientInsuranceNumber: String): PatientDtoOut? =
         patientRepository.getAndMapPatientsBy {
             Patients.insuranceNumber eq patientInsuranceNumber.trim()
         }.singleOrNull()?.withSortedAnswers()
-            ?: throw entityNotFound<Patients>(Patients::insuranceNumber, patientInsuranceNumber)
+
+    /**
+     * Returns single patient with given personal or insurance number or throws exception.
+     */
+    @Suppress("MagicNumber") // 11 is the maximal length of a Czech personal number
+    suspend fun getPatientByPersonalOrInsuranceNumber(patientPersonalOrInsuranceNumber: String): PatientDtoOut {
+        // TODO write Kotlin...
+        // TODO theoretically, there could be a patient with an insurance number same as personal number of some other patient.
+        //  In this case, we cannot find the patient with the personal number.
+        var patient = getPatientByInsuranceNumber(patientPersonalOrInsuranceNumber)
```

Comment:
nooo takze tady teda delame dve query jo? neni snazsi pouzit ten union co jsem ti posilal?

(ve vysledku je to asi sumak, tuhle "neoptimalitu" si muzeme dovolit)

Reply:
No to zalezi. Pokud se ptas: Neni pro tebe snazsi naprogramovat to s tim unionem? Tak ne, neni :D. Pokud se ptas: Nebylo by to hezci, elegantnejsi, prehlednejsi a pro nekoho urcite snazsi? Pak ano :D